typescript/infra/src/funding/adapters/SealevelFundingAdapter.ts

```diff
@@ -0,0 +1,303 @@
+import {
+  ComputeBudgetProgram,
+  LAMPORTS_PER_SOL,
+  PublicKey,
+  SystemProgram,
+  Transaction,
+  sendAndConfirmTransaction,
+} from '@solana/web3.js';
+import { Logger } from 'pino';
+import { Gauge } from 'prom-client';
+
+import {
+  BaseSealevelAdapter,
+  ChainMap,
+  ChainName,
+  MultiProtocolProvider,
+  SEALEVEL_PRIORITY_FEES,
+  SealevelIgpAdapter,
+} from '@hyperlane-xyz/sdk';
+
+import { Contexts } from '../../../config/contexts.js';
+import { BaseAgentKey } from '../../agents/keys.js';
+import { DeployEnvironment } from '../../config/environment.js';
+import { FundableRole } from '../../roles.js';
+import { FundingAddresses } from '../types.js';
+
+import { IFundingAdapter } from './IFundingAdapter.js';
+
+/**
+ * Sealevel-specific implementation of the funding adapter
+ */
+export class SealevelFundingAdapter
+  extends BaseSealevelAdapter
+  implements IFundingAdapter
+{
+  private igpAdapter: SealevelIgpAdapter;
+  private igpProgramIds: ChainMap<string>;
+
+  constructor(
+    chainName: ChainName,
+    multiProvider: MultiProtocolProvider,
+    environment: DeployEnvironment,
+    private readonly context: Contexts,
+    private readonly fundingAddresses: FundingAddresses,
+    public readonly logger: Logger,
+    private readonly walletBalanceGauge: Gauge<string>,
+  ) {
+    super(chainName, multiProvider, fundingAddresses, logger);
+    this.logger = logger;
+    // TODO: this should be a config that is passed in
+    this.igpProgramIds = {
+      solanamainnet: 'BhNcatUDC2D5JTyeaqrdSukiVFsEHK7e3hVmKMztwefv',
+      sonicsvm: '3kd3K1yymsfYoqzNnktLwpKXKeB9Y2PRf9rfb31k4o13',
+      soon: '5AAMBemjUrAprKJzi22Si9pyYhK8vTr7Jpti7LEXo6d8',
+      eclipsemainnet: 'Hs7KVBU67nBnWhDPZkEFwWqrFMUfJbmY2DQ4gmCZfaZp',
+    };
+    this.igpAdapter = new SealevelIgpAdapter(
+      this.chainName,
+      this.multiProvider,
+      {
+        igp: this.fundingAddresses.interchainGasPaymaster,
```

Comment:
asked earlier but if igp is the only thing in fundingAddresses - is it even needed? I believe mpp/mp extend the chain metadata manager or something so should have all the addresses+metadata we need 

ie getChainMetadata(chain).interchainGasPaymaster or something like that, but could be wrong

Reply:
ditto in the other adapter